corpus/impulse/forms.py

```diff
@@ -0,0 +1,57 @@
+from impulse.models import ImpulseUser, Team, Announcement
+
+from corpus.forms import CorpusModelForm
+
+class ImpulseForm(CorpusModelForm):
+    class Meta:
+        model = ImpulseUser
+        fields = [
+            "from_nitk",
+            "college_name",
+            "roll_no",
+            "ieee_member",
+            "ieee_membership_no",
+        ]
+    
+    def __init__(self, *args, **kwargs):
+        super(ImpulseForm, self).__init__(*args, **kwargs)
+        # set a required attribute based on conditions
+        self.fields["college_name"].required = not self.fields['from_nitk'].widget.value_from_datadict(
+            self.data, self.files, self.add_prefix('from_nitk')
+        )
+        self.fields["roll_no"].required = self.fields['from_nitk'].widget.value_from_datadict(
+            self.data, self.files, self.add_prefix('from_nitk')
+        )
+        self.fields["ieee_membership_no"].required = self.fields['ieee_member'].widget.value_from_datadict(
+            self.data, self.files, self.add_prefix('ieee_member')
+        )
```

Comment:
You can do these checks in the `clean` function in forms. Refer to https://github.com/IEEE-NITK/corpus/blob/main/corpus/embedathon/forms.py#L27

Reply:
Done, implemented clean function.